### ISSUE TITLE
Default given value doesn't work inside given definition 

### ISSUE THREAD

**Issue by t9dupuy** [scala/scala3#20088]
## Compiler version

`3.4.0`

## Minimized code

```Scala
trait Foo
trait Bar

given (using foo: Foo = new {}): Bar with {}

summon[Bar]
```

## Output

```
No given instance of type Bar was found for parameter x of method summon in object Predef.
I found:

    given_Bar(using )

But given instance given_Bar does not match type Bar.
```

## Expectation

Minimized code should compile

*Alternatively*, compiler should prevent (or at least warn) user from giving context parameters default values inside given definition.

**Comment by mbovel** [comment]
This might be related to #19414. If we tweak lookup of implicits with default arguments, maybe this could also fix this issue.

**Comment by mbovel** [comment]
While the reporting would be improved by the fix of #19414, the core issue is different; it's related to the designing of `with` when there are default arguments:

```
trait Foo
trait Bar
given wrong(using foo: Foo = new {}): Bar with {}
given good(using foo: Foo): Bar with {}
```

```scala
[[syntax trees at end of                     typer]] // tests/neg/20088.scala
package <empty> {
  ...
  final module class 20088$package() extends Object() {
    ...
    given class wrong(using foo: Foo) extends Object(), Bar { ... }
    final lazy module val wrong: wrong = new wrong()
    final module class wrong() extends AnyRef() { this: wrong.type => ... }
    final given def wrong(using foo: Foo): wrong = new wrong(using foo)()

    given class good(using foo: Foo) extends Object(), Bar { ... }
    final given def good(using foo: Foo): good = new good(using foo)()
  }
}
```

**Comment by mbovel** [comment]
Actually, the name clash is not a problem. The issue is that the lookup for default argument fails: the default parameters getters are associated to the constructor, not to the generated implicit wrapper, and are not found.
